src/main/java/org/javarosa/core/model/ItemsetBinding.java

```diff
@@ -191,15 +191,20 @@ private SelectChoice getChoiceForTreeReference(FormDef formDef, DataInstance for
         // Provide a default value if none is specified
         value = value != null ? value : "dynamic:" + i;
 
-        TreeElement copyNode = null;
+        SelectChoice choice = new SelectChoice(
+            label,
+            value,
+            labelIsItext,
+            item.getInstanceName() != null
+                ? (TreeElement) formDef.getNonMainInstance(item.getInstanceName()).resolveReference(item)
+                : formDef.getMainInstance().resolveReference(item));
+
+        choice.setIndex(i);
+
         if (copyMode) {
```

Comment:
This `copyMode` stuff isn't tested or documented. I think we should remove all of it in the near-ish future. For now I wanted to at least group it together so it's out of the way.